titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/event/TaskUpdateEvent.java

```diff
@@ -46,43 +53,51 @@ public Task getCurrentTask() {
         return previousTask;
     }
 
+    public boolean isMovedFromAnotherJob() {
+        return movedFromAnotherJob;
+    }
+
     @Override
     public boolean equals(Object o) {
         if (this == o) {
             return true;
         }
-        if (o == null || getClass() != o.getClass()) {
+        if (!(o instanceof TaskUpdateEvent)) {
             return false;
         }
         if (!super.equals(o)) {
             return false;
         }
-
         TaskUpdateEvent that = (TaskUpdateEvent) o;
-
-        return currentJob != null ? currentJob.equals(that.currentJob) : that.currentJob == null;
+        return movedFromAnotherJob == that.movedFromAnotherJob &&
+                currentJob.equals(that.currentJob) &&
+                currentTask.equals(that.currentTask) &&
+                previousTask.equals(that.previousTask);
     }
 
     @Override
     public int hashCode() {
-        int result = super.hashCode();
-        result = 31 * result + (currentJob != null ? currentJob.hashCode() : 0);
-        return result;
+        return Objects.hash(super.hashCode(), currentJob, currentTask, previousTask, movedFromAnotherJob);
     }
 
     @Override
     public String toString() {
         return "TaskUpdateEvent{" +
                 "currentJob=" + currentJob +
-                ", current=" + getCurrent() +
-                ", previous=" + getPrevious() +
-                "}";
+                ", currentTask=" + currentTask +
```

Comment:
The old `toString` used to call the super classes methods to print out information.

Reply:
I don't think so. Maybe you are reffering to `equals()` and `hashcode()`? The super class doesn't even have a `toString`, and our current template prints out the class type, which would be awkward when calling from a subclass.

Reply:
I will remove the duplicated fields and use the superclass for all of it.